### PR TITLE
[Community] Experimental AnimateDiff Image to Video (open to improvements)

### LINKED ISSUE #6123: AnimateDiffPipeline: add ImageToVideo and VideoToVideo
Can adapt the work in https://github.com/Kosinkadink/ComfyUI-AnimateDiff-Evolved

### PR DESCRIPTION
**Pull request by a-r-r-o-w**
# What does this PR do?

Experimental img2vid support for AnimateDiff. Moved it here from #6328 due to the other being a core addition and it should make it easier to review vid2vid.

[Colab](https://colab.research.google.com/drive/1TvzCDPHhfFtdcJZe4RLloAwyoLKuttWK?usp=sharing)

Fixes #6123.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs?
- [x] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@DN6

### CODE REVIEW

**Comment by a-r-r-o-w** [comment]
Some results can be found here: https://github.com/huggingface/diffusers/pull/6328#issuecomment-1869900042

**Review by DN6 — APPROVED** [review]
Excellent 👍🏽

File: examples/community/README.md [hunk 1]
```diff
@@ -3279,7 +3281,7 @@ export_to_video(
 
 ### StyleAligned Pipeline
 
-This pipeline is the implementation of [Style Aligned Image Generation via Shared Attention](https://arxiv.org/abs/2312.02133).
+This pipeline is the implementation of [Style Aligned Image Generation via Shared Attention](https://arxiv.org/abs/2312.02133). Checkout [this](https://github.com/huggingface/diffusers/pull/6489) PR for more examples and results.
```
Comment (DN6) [inline examples/community/README.md:3284]: Was this just added?

Comment (a-r-r-o-w) [reply examples/community/README.md:3284]: Yes. I thought it'd be okay since I forgot doing it on the StyleAligned PR. Should I remove it?

Comment (DN6) [reply examples/community/README.md:3284]: Yes please. Then we can merge 👍🏽

Comment (a-r-r-o-w) [reply examples/community/README.md]: hi @DN6, done. lgtm!

**Review by DN6 — APPROVED** [review]
Nice!

File: examples/community/README.md [hunk 2]
```diff
@@ -3409,6 +3411,29 @@ images = pipe(
...
+image = load_image("snail.png")
+output = pipe(
+  image=image,
+  prompt="A snail moving on the ground",
+  strength=0.8,
+  latent_interpolation_method="slerp", # can be lerp, slerp, or your own callback
+)
+frames = output.frames[0]
+export_to_gif(frames, "animation.gif")
+=======
```
Comment (patrickvonplaten) [inline examples/community/README.md:3436]:
```suggestion
```

Comment (a-r-r-o-w) [reply examples/community/README.md:3436]: Looks like I messed up in the merge conflict :sweat_smile:

**Comment by a-r-r-o-w** [comment]
image

Looks like the IP Adapter FaceID PR would require fixing in the community README. @DN6 Should I fix it here?

**Comment by DN6** [comment]
@a-r-r-o-w  I think separate PR for the README works.

**Comment by Skquark** [comment]
I ran into an error using pipeline_animatediff_img2video worth reporting.. Getting this import error:
```
File "/root/.cache/huggingface/modules/diffusers_modules/git/pipeline_animatediff_img2video.py", line 28, in <module>
    from diffusers.models.unet_motion_model import MotionAdapter
ModuleNotFoundError: No module named 'diffusers.models.unet_motion_model'
```
To fix it, you'd change that line to:
`from diffusers.models.unets.unet_motion_model import MotionAdapter`
Looks like it was missing the .unets part as I see in animatediff_controlnet.py.. Tested in my custom at "AlanB/pipeline_animatediff_img2video_mod" and got it working in DiffusionDeluxe.com app.  I also notice the output was grainy, thinking the DDIM scheduler code in the example may be incorrect to, gonna play with that some more to make sure. Thanks.

**Comment by DN6** [comment]
cc: @a-r-r-o-w

**Comment by a-r-r-o-w** [comment]
@Skquark Thanks for reporting the error! I believe this PR overlapped with the refactoring of the UNet modules and so this broke. If you'd like to, you can submit a PR fixing it otherwise I'll take it up sometime soon. Regarding the quality, AnimateDiff motion checkpoints are sensitive to the `beta_schedule="linear"` parameter so make sure that you set it when initializing the scheduler. The output quality here is not that great because of how the latents are created from an image (it is a method that the community figured out that produces ~handwavy results). If you're looking into it more deeply to improve result quality, maybe [this](https://github.com/huggingface/diffusers/pull/6328#issuecomment-1869919269) comment can provide some useful information. I'd love if you can figure out how generation quality could be improved, and congrats on building what you have! But, I do not think much can be done to improve generation quality with a hacky trick like how the latents are created here, or atleast don't know of any such technique (maybe @jon-chuang could share his thoughts), and so I'd recommend using a model geared towards img2vid such as [Personalized Image Animator](https://huggingface.co/docs/diffusers/en/api/pipelines/pia) or [MotionCtrl SVD](https://github.com/huggingface/diffusers/pull/7005).

**Comment by Skquark** [comment]
Thanks, simple enough fix for an understandable refactoring break. I'd prefer it if you or someone else does that PR since it's an easy one-liner and my git fork is a mess, but I can try if preferred.. I also changed the way I loaded the scheduler to be the same as the video2video example using .from_pretrained, and it worked much better without the defective graininess, so may want to change that too in the usage example.. Worked right with this setting:
`pipe.scheduler = DDIMScheduler.from_pretrained(model_id, subfolder="scheduler", clip_sample=False, timestep_spacing="linspace", beta_schedule="linear", steps_offset=1)`
instead of DDIMScheduler(beta_schedule="linear".... Another easy fix, got some pretty fun test results so far, looking forward to see what we can do with it next. I've got a nice collection of most of the good opensource Video AIs in my app, including PIA, I2VGen-XL, Video-to-Video, Controlnet Vid2Vid, SVD, Enhanced AnimateDiff from cli mod instead of HF, and all the others I could find.  Will add MotionCtrl SVD when it's merged, and more as they come, they're all fun to play with... Now I'm anxious to see what SD3 will do for videos in the next leap. Anyways, here's my screenshots of this pipeline's implementation..
SDD_AnimateDiff_Img2Vid-1
SDD_AnimateDiff_Img2Vid-2

**Pull request merged**